### PR TITLE
docs: docs for new KEY_FORMAT property and associated server configs

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

Porting over https://github.com/confluentinc/ksql/pull/6252 to a branch on the main repo rather than my personal fork, in preparation for working on this jointly with @big-andy-coates .

Opening a separate docs PR for the changes in https://github.com/confluentinc/ksql/pull/6194 as these docs cannot go live before the feature flag guarding https://github.com/confluentinc/ksql/pull/6194 is enabled by default.

### Testing done 

Docs-only change.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Review by JimGalasyn — APPROVED** [review]
Looks great!

File: docs/developer-guide/ksqldb-reference/create-stream-as-select.md [hunk 1]
```diff
@@ -75,7 +75,9 @@ The WITH clause for the result supports the following properties:
...
 | ----------------- | ---------------------------------------------------------------------------------------------------- |
 | KAFKA_TOPIC       | The name of the Kafka topic that backs this stream. If this property is not set, then the name of the stream in upper case will be used as default. |
-| VALUE_FORMAT      | Specifies the serialization format of the message value in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format of the input stream/table is used. |
+| KEY_FORMAT        | Specifies the serialization format of the message key in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the input stream/table is used. |
+| VALUE_FORMAT      | Specifies the serialization format of the message value in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the input stream/table is used. |
```
Comment (big-andy-coates) [inline docs/developer-guide/ksqldb-reference/create-stream-as-select.md:79]:
```suggestion
| KEY_FORMAT        | Specifies the serialization format of the message key in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the left-most input stream/table is used. |
| VALUE_FORMAT      | Specifies the serialization format of the message value in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the left-most input stream/table is used. |
```

File: docs/developer-guide/ksqldb-reference/create-table-as-select.md [hunk 2]
```diff
@@ -79,7 +79,9 @@ The WITH clause supports the following properties:
...
 | ----------------- | ---------------------------------------------------------------------------------------------------- |
 | KAFKA_TOPIC       | The name of the Kafka topic that backs this table. If this property is not set, then the name of the table will be used as default. |
-| VALUE_FORMAT      | Specifies the serialization format of the message value in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, then the format of the input stream/table is used. |
+| KEY_FORMAT        | Specifies the serialization format of the message key in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the input stream/table is used. |
+| VALUE_FORMAT      | Specifies the serialization format of the message value in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the input stream/table is used. |
```
Comment (big-andy-coates) [inline docs/developer-guide/ksqldb-reference/create-table-as-select.md:83]:
```suggestion
| KEY_FORMAT        | Specifies the serialization format of the message key in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the left-most input stream/table is used. |
| VALUE_FORMAT      | Specifies the serialization format of the message value in the topic. For supported formats, see [Serialization Formats](../serialization.md#serialization-formats). If this property is not set, the format from the left-most input stream/table is used. |
```

File: docs/operate-and-deploy/installation/server-config/config-reference.md [hunk 3]
```diff
@@ -282,6 +282,29 @@ The corresponding environment variable in the
...
 is `KSQL_KSQL_FUNCTIONS_SUBSTRING_LEGACY_ARGS`.
 
+### ksql.persistence.default.format.key
+
+Sets the default value for the `KEY_FORMAT` property if one is
+not supplied explicitly in [CREATE TABLE](../../../developer-guide/ksqldb-reference/create-table.md)
+or [CREATE STREAM](../../../developer-guide/ksqldb-reference/create-stream.md)
+statements.
+
+The default value for this configuration is `KAFKA`.
```
Comment (big-andy-coates) [inline docs/operate-and-deploy/installation/server-config/config-reference.md:292]:
```suggestion
The default value for this configuration is `KAFKA`.

If not set and no explicit format is provided in the statement, via either the `KEY_FORMAT` or the `FORMAT` property, the statement will be rejected as invalid.
```

File: docs/operate-and-deploy/installation/server-config/config-reference.md [hunk 4]
```diff
@@ -282,6 +282,29 @@ The corresponding environment variable in the
...
+
+### ksql.persistence.default.format.value
+
+Sets the default value for the `VALUE_FORMAT` property if one is
+not supplied explicitly in [CREATE TABLE](../../../developer-guide/ksqldb-reference/create-table.md)
+or [CREATE STREAM](../../../developer-guide/ksqldb-reference/create-stream.md)
+statements.
+
+If not set and no explicit value is provided in the statement, via either the `VALUE_FORMAT` or the
+`FORMAT` property, an error will be thrown.
```
Comment (big-andy-coates) [inline docs/operate-and-deploy/installation/server-config/config-reference.md:304]:
```suggestion
If not set and no explicit value is provided in the statement, via either the `VALUE_FORMAT` or the
`FORMAT` property, the statement will be rejected as invalid.
```

**Review by big-andy-coates — COMMENTED** [review]
@vcrfxia, it's probably worth updating `craete-a-table.md` and `create-a-stream.md` as well.  I

File: docs/operate-and-deploy/installation/server-config/config-reference.md (hunk 4 above)
Comment (vcrfxia) [reply docs/operate-and-deploy/installation/server-config/config-reference.md:304]: Ah sorry, not sure how I lost this comment from your last round of reviews. Fixed!

**Review by vcrfxia — COMMENTED** [review]
> it's probably worth updating craete-a-table.md and create-a-stream.md as well.

Which update did you have in mind when you wrote this? The changes made to `create-stream-as-select.md` and `create-table-as-select.md` don't apply to CS/CT statements since there's no concept of inputs there.

**Comment by big-andy-coates** [comment]
> > it's probably worth updating craete-a-table.md and create-a-stream.md as well.
> 
> Which update did you have in mind when you wrote this? The changes made to `create-stream-as-select.md` and `create-table-as-select.md` don't apply to CS/CT statements since there's no concept of inputs there.

You've made doc changes to `create-stream.md` and `create-table.md`, adding in the `KEY_FORMAT` etc.  I'm suggesting `create-a-table.md` and `create-a-stream.md` would also benefit from covering using `KEY_FORMAT`, `FORMAT` and the new configs.

**Pull request merged**